src/bindings/scala/api/src/main/scala/com/ctc/omega_edit/SessionImpl.scala

```diff
@@ -208,9 +208,12 @@ private[omega_edit] class SessionImpl(p: Pointer, i: FFI) extends Session {
     try {
       val result = i.omega_session_get_segment(p, sp, offset)
 
-      Option.when(result == 0)(
-        Segment(offset, i.omega_segment_get_data(sp).getBytes)
-      )
+      Option.when(result == 0) {
+        val data = i.omega_segment_get_data(sp)
+        val out = Array.ofDim[Byte](length.toInt)
+        data.get(0, out, 0, length.toInt)
+        Segment(offset, out)
+      }
```

Comment:
This got uglier.

Reply:
Yeah I was surprised there wasn't a built-in to do this. It seems to either give you a `String` (using the null-termination) or you have to use a `Pointer` yourself.